nemo_skills/evaluation/evaluator/ojbench.py

```diff
@@ -0,0 +1,135 @@
+        LOG.info("Installing required packages for ojbench evaluation...")
+
+        clone_cmd = "git clone https://github.com/He-Ren/OJBench.git"
+        result, _ = await sandbox.execute_code(clone_cmd, language="shell", timeout=300)
+        if result["process_status"] != "completed":
+            LOG.warning(f"Failed to clone OJBench repo: {result.get('stderr', 'Unknown error')}")
+            return False
+
+        install_cmd = "pip install -e OJBench"
+        result, _ = await sandbox.execute_code(install_cmd, language="shell", timeout=300)
+        if result["process_status"] != "completed":
+            LOG.warning(f"Failed to install ojbench: {result.get('stderr', 'Unknown error')}")
+            return False
+
+        LOG.info("Successfully installed ojbench.")
+        return True
+
+
+async def eval_ojbench_async(cfg):
+    eval_config = OJBenchConfig(**cfg.eval_config)
+    problem_dirs = [
+        Path(cfg.data_dir, "ojbench/OJBench_testdata/NOI"),
+        Path(cfg.data_dir, "ojbench/OJBench_testdata/ICPC"),
+    ]
+
+    if not await install_packages(eval_config):
+        return
+
+    async with sandbox_context(eval_config.sandbox) as sandbox:
+        for jsonl_file in unroll_files(cfg.input_files):
+            with open(jsonl_file, encoding="utf-8") as f_in:
+                samples = []
+                for line in f_in:
+                    sample = json.loads(line)
+                    sample = preprocess_code(sample, sample["language"], strip_whitespace=True)
+                    sample["prompt"] = sample.pop("question")
+                    sample["content"] = f"```{sample['language']}\n{sample['completion']}\n```"
+                    sample.pop("completion")
+                    samples.append(sample)
+
+            with open(jsonl_file, "w", encoding="utf-8") as f_out:
+                f_out.writelines(json.dumps(sample) + "\n" for sample in samples)
+
+            eval_results_path = jsonl_file[:-6] + "_eval_results.json"
+            LOG.info(f"jsonl_file = {jsonl_file}.")
+            LOG.info(f"eval_results_path = {eval_results_path}.")
+            eval_code = textwrap.dedent(f"""
+                import ojbench
+                ojbench.init(problem_dirs={repr([str(p) for p in problem_dirs])})
+                ojbench.judge_jsonl(
+                    input_path={repr(jsonl_file)},
+                    output_path={repr(eval_results_path)},
+                    num_workers=16
+                )
+            """)
+
+            cmd = f'env -i PATH="/usr/local/bin:/usr/bin:/bin" python3 -c {shlex.quote(eval_code)}'
+            output, _ = await sandbox.execute_code(
+                cmd,
+                language="shell",
+                timeout=eval_config.timeout * len(samples) + 60,
+                max_output_characters=100_000,
+            )
+
+            if output.get("process_status") != "completed":
+                LOG.error(f"Evaluation failed for {jsonl_file}. Stderr: {output.get('stderr')}")
```

Comment:
what's the point of skipping this? Let's just fail with error, otherwise it might lead to hard-to-notice issues